src/backend/distributed/planner/multi_planner.c

```diff
@@ -667,6 +624,30 @@ CheckNodeIsDumpable(Node *node)
 
 
 /*
+ * CheckNodeCopyAndSerialization checks copy/dump/read functions
+ * for nodes.
+ *
+ * It does not check string equality on node dumps due to differences
+ * in some Postgres types.
+ */
+static void
+CheckNodeCopyAndSerialization(Node *node)
```

Comment:
I think we should return the copy, and use that - that enforces the copy is actually correct.